### DISCUSSION TITLE
Proper way of nesting components with Alpine.

### DISCUSSION THREAD

**Discussion by eddysantos** [livewire/livewire#3857] [1. Help]
Hello everyone, 

What would be the proper way to nest Livewire components, that rely heavily on Alpine? In said nested components I'm extracting alpine logic into a function, however, whenever I add the nested component, the function is not recognized. 

Something with mentioning is that while the component is nested, is not loaded on the initial page load, and this may have something to do with the fact that it's not working. This is how I'm including the component: 

```
@if ($selected_trip_detail)
     @livewire('operations.details.operation-detail', ['linehaul_id' => $selected_trip_detail])
@endif
```

Thanks in advance for your help!!

**Comment by joshhanley** [comment]
@eddysantos yeah scripts aren't currently loaded dynamically, so it depends on how you are including the function.

From the next version of Livewire, you will be able to push your function to a scripts stack using `@push('scripts')` and then the Alpine function will be loaded dynamically as expected.

Hope this helps!
